dask/dataframe/partitionquantiles.py

```diff
@@ -285,13 +291,13 @@ def merge_and_compress_summaries(vals_and_weights):
     weights_append = weights.append
     val, weight = prev_val, prev_weight = next(it)
     for val, weight in it:
-        if val == prev_val:
+        if val == prev_val or val is prev_val:
```

Comment:
Did we run into a case where `val != prev_val` but `val is prev_val`?

Reply:
Think this may be a remnant of when `val` could've been `pd.NA`, as checking equality against `pd.NA` just returns `pd.NA`; think we should be good to remove now